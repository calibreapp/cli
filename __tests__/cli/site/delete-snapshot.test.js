const {
  runCLI,
  setupIntegrationServer,
  teardownIntegrationServer
} = require('../../utils')
const deleteSnapshot = require('../../fixtures/deleteSnapshot.json')

describe('site delete-snapshot', () => {
  beforeAll(() => setupIntegrationServer(deleteSnapshot))
  afterAll(() => teardownIntegrationServer())

  test('requires id', () => {
    return runCLI({
      args: 'site delete-snapshot --site=test',
      testForError: true
    }).then(stdout => {
      expect(stdout).toMatchSnapshot()
    })
  })

  it('deletes snapshot', () => {
    return runCLI({
      args: 'site delete-snapshot --site=test --id=1',
      testForError: true
    }).then(stdout => {
      expect(stdout).toMatchSnapshot()
    })
  })
})
