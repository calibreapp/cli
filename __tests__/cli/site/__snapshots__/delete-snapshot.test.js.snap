// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`site delete-snapshot deletes snapshot 1`] = `
"- Connecting to Calibre
âœ” Snapshot deleted: 14"
`;

exports[`site delete-snapshot requires id 1`] = `
"

 Missing required argument: id 

 --- Command documentation
 calibre site delete-snapshot [options]

Deletes a snapshot from a site

Options:
  --help     Show help                                                 [boolean]
  --version  Show version number                                       [boolean]
  --id       The id of the snapshot                                   [required]
  --site     The identifying slug of a site                  [string] [required]
  --confirm  Confirm the deletion
  --json     Return the output in JSON format"
`;
